code/game/gamemodes/cult/runes.dm

```diff
@@ -213,17 +210,30 @@ structure_check() searches for nearby cultist structures required for the invoca
 	talisman_type = possible_talismans[entered_talisman_name]
 	if(!Adjacent(user) || !src || qdeleted(src) || user.incapacitated() || rune_in_use || !talisman_type)
 		return
+	papers_on_rune = checkpapers()
+	if(!papers_on_rune.len)
+		user << "<span class='cultitalic'>There must be a blank paper on top of [src]!</span>"
+		fail_invoke()
+		log_game("Talisman Creation rune failed - no blank papers on rune")
+		return
+	var/obj/item/weapon/paper/paper_to_imbue = pick(papers_on_rune)
```

Comment:
If you dont need to differentiate between the members of a list(all are the same empty papers) list[1] is better than pick()
Microoptimizations are inportant